content/blog-categories/old-tech-blog/w3c/contents.lr

```diff
@@ -0,0 +1 @@
+name: w3c
```

Comment:
Do these files need to be created manually whenever a new category is created? How do we ensure that that happens?

Reply:
Yes, we have to manually add it either through the admin panel or in the folder structure. I checked the [official documentation](https://www.getlektor.com/docs/guides/categories/) and their example is also implemented this way. 
I will look into it and update you about it.

Reply:
@kgodey We can make categories of `type=checkboxes` instead of `strings` to make sure that we have created a category before we can use it and also we can still mark multiple categories. 

But I am facing a problem in making the `source` dynamic.

If we have for example two blogs, `foo` and `bar`. Then there will be `blog-categories/foo/*` and `blog-categories/bar/*` directories to hold respective categories.
In `blog-post.ini` model:
```
[fields.categories]
label = Categories
type = checkboxes
source = site.query(<dynamic source=?>)
```
I have tried `source = site.query('/blog-categories/' + this.parent.path.split('/')[-1])`, but it is producing `this is undefined` error. Infact, I think we cannot use any variables in user defined model fields but I may be wrong. 
I have searched through the documentation and couldn't find any solution.
Is there any way we can solve this problem? 
Thanks

Reply:
How about moving `blog-categories` into the `old-tech-blog` parent folder (I would probably just rename it to `categories`) since the categories are associated with that blog. You could move the blog entries into their own folder if needed (for example, see the `project-ideas-collection` folder under `gsoc-2019`) so that the top level folders would just be `categories`, `entries`, and (later) `authors`. You should be able to build the path from that.

Reply:
Thanks, @kgodey  for this recommendation. I have tried this locally and everything works well. The structure is cleaner this way.
I am going to test it more by adding dummy blogs.
Shall I make another PR for this as significant editing and restructuring are happening?

Also, I think we will only be able to set `type=strings` for `[fields.categories]` under `blog-post.ini` model. For `type=checkboxes`, all categories must reside under a common path(which is not happening in this implementation) as demonstrated in [this lektor guide](https://www.getlektor.com/docs/guides/categories/). Everything will work normally, but we still have to make sure that we make `foo` category file under `<blog-name>/categories` folder for the category link `foo` in a blog post to work properly.
What are your thoughts on this?
Thanks again!

Reply:
I think using `type=strings` is fine. Would you be able to add a check to the Lektor build process so that there's an error if there's a missing category page?

Feel free to make another PR; I don't have an opinion either way.

Reply:
@kgodey One way to add a check to the Lektor build is to build a custom plugin. It will live under `packages/` directory(which is not present currently).
I have made a [simple plugin](https://gist.github.com/makkoncept/9cc294c2fd7bce600e0f8df8b5149d2d) to inject a function `check_file(filename)` into template globals which can be used to raise an error if the corresponding categories do not exist. Same can be done in implementing `authors` if needed.
Shall I continue this way?

Reply:
Yes, that approach sounds good, thanks!